template is sanitized by Angular

```
{
      key: "test",
      template: `test: <input>`,
      templateOptions: {
        label: "test"
      }
    }
```
 WARNING: sanitizing HTML stripped some content, see http://g.co/ng/security#xss

this causes that <input> not being rendered.

> use `sanitizer` service:
> 
> ```ts
> import { DomSanitizer } from '@angular/platform-browser';
> ...
> 
> fields: = [{
>   key: "test",
>   template: this.sanitizer.bypassSecurityTrustHtml(`test: <input>`),
>   templateOptions: {
>     label: "test"
>   }
> }];
> 
> constructor(private sanitizer: DomSanitizer) {}
> ```

keep in mined that this method will cause error TS2322: Type 'SafeHtml' is not assignable to type 'string'.

--- a/src/core/src/lib/templates/field-template.type.ts
+++ b/src/core/src/lib/templates/field-template.type.ts
@@ -1,8 +1,27 @@
 import { Component } from '@angular/core';
+import { DomSanitizer } from '@angular/platform-browser';
 import { FieldType } from './field.type';
 
 @Component({
   selector: 'formly-template',
-  template: `<div [innerHtml]="field.template"></div>`,
+  template: `<div [innerHtml]="template"></div>`,
 })
-export class FormlyTemplateType extends FieldType {}
+export class FormlyTemplateType extends FieldType {
+  get template() {
+    if (this.field && (this.field.template !== this.innerHtml.template)) {
+      this.innerHtml = {
+        template: this.field.template,
+        content: this.to.safeHtml
+          ? this.sanitizer.bypassSecurityTrustHtml(this.field.template)
+          : this.field.template,
+      };
+    }
+
+    return this.innerHtml.content;
+  }
+
+  private innerHtml = { content: null, template: null };
+  constructor(private sanitizer: DomSanitizer) {
+    super();
+  }
+}

